File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/UnknownReactionException.java
package com.github.dedis.popstellar.utility.error;

import com.github.dedis.popstellar.R;

public class UnknownReactionException extends GenericException {

  public UnknownReactionException() {
    super("Deleting a reaction which is unknown");
  }

  @Override
  public int getUserMessage() {
    return R.string.unknown_reaction_exception;
  }

  @Override
  public Object[] getUserMessageArguments() {
    return new Object[0];
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/UnknownReactionException.kt
package com.github.dedis.popstellar.utility.error

import com.github.dedis.popstellar.R

class UnknownReactionException : GenericException("Deleting a reaction which is unknown") {
    override val userMessage: Int
        get() = R.string.unknown_reaction_exception
    override val userMessageArguments: Array<Any?>
        get() = arrayOfNulls(0)
}